libs/usb-mocking/insert.sh

```diff
@@ -0,0 +1,20 @@
+DIR=$( dirname $BASH_SOURCE )
+
+if [ -f "$DIR/usb-mock.img" ]; then
+    echo "Virtual USB drive found..."
+else
+    echo "No virtual USB drive found. Initializing a virtual USB drive with default size of 64MB..."
+    bash $DIR/initialize.sh -s 64
+    echo "To create a larger virtual USB drive, run 'sudo bash initalize.sh -s <size>' where <size> is the desired size in megabytes."
+fi
+
+# Attach the loop device
+DEVICE_BASENAME=$( basename $( losetup -fP --show $DIR/usb-mock.img ) )
+
+# Use a delay to allow the OS to detect the filesystems on the loop device partition
+sleep 1 
```

Comment:
Is there a way to know when the OS has detected the filesystems? E.g. a command you could run to check? If so, maybe you could loop here to wait for that

Reply:
Yeah, that's doable!